[qemu] Unable to build

```
block/ssh.c:301:48: error: ‘SSH_PUBLICKEY_HASH_SHA256’ undeclared (first use in this function); did you mean ‘SSH_PUBLICKEY_HASH_SHA1’?
             r = ssh_get_publickey_hash(pubkey, SSH_PUBLICKEY_HASH_SHA256,
                                                ^~~~~~~~~~~~~~~~~~~~~~~~~
                                                SSH_PUBLICKEY_HASH_SHA1
block/ssh.c:301:48: note: each undeclared identifier is reported only once for each function it appears in
block/ssh.c:307:27: warning: implicit declaration of function ‘ssh_get_fingerprint_hash’; did you mean ‘ssh_get_pubkey_hash’? [-Wimplicit-function-declaration]
             fingerprint = ssh_get_fingerprint_hash(SSH_PUBLICKEY_HASH_SHA256,
                           ^~~~~~~~~~~~~~~~~~~~~~~~
                           ssh_get_pubkey_hash
block/ssh.c:307:27: warning: nested extern declaration of ‘ssh_get_fingerprint_hash’ [-Wnested-externs]
make: *** [/usr/local/tmp/crew/qemu-4.2.0.tar.xz.dir/qemu-4.2.0/rules.mak:69: block/ssh.o] Error 1
make: *** Waiting for unfinished jobs....
qemu failed to build: `make V=0 -j2` exited with 2
```

Hello I had the same problem, but I was able to install some dependencies, and now my bug looks like this:


../lib64/libgtk-3.so: undefined reference to `hb_ft_font_set_funcs'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:208: qemu-system-arm] Error 1
make[1]: Leaving directory '/usr/local/tmp/crew/qemu-4.2.0.tar.xz.dir/qemu-4.2.0/arm-softmmu'
make: *** [Makefile:491: arm-softmmu/all] Error 2
/usr/local/lib64/gcc/x86_64-cros-linux-gnu/8.3.0/../../../../x86_64-cros-linux-gnu/bin/ld: /usr/lib64/libfontconfig.so.1: unknown type [0x13] section `.relr.dyn'
/usr/local/lib64/gcc/x86_64-cros-linux-gnu/8.3.0/../../../../x86_64-cros-linux-gnu/bin/ld: /usr/local/lib64/libcrypto.so.46: warning: EVP_EncryptFinal is often misused, please use EVP_EncryptFinal_ex and EVP_CIPHER_CTX_cleanup
/usr/local/lib64/gcc/x86_64-cros-linux-gnu/8.3.0/../../../../x86_64-cros-linux-gnu/bin/ld: /usr/local/lib64/libcrypto.so.46: warning: EVP_DecryptFinal is often misused, please use EVP_DecryptFinal_ex and EVP_CIPHER_CTX_cleanup
/usr/local/lib64/gcc/x86_64-cros-linux-gnu/8.3.0/../../../../x86_64-cros-linux-gnu/bin/ld: /usr/local/lib64/libcrypto.so.46: warning: EVP_CipherFinal is often misused, please use EVP_CipherFinal_ex and EVP_CIPHER_CTX_cleanup
/usr/local/lib64/gcc/x86_64-cros-linux-gnu/8.3.0/../../../../x86_64-cros-linux-gnu/bin/ld: /usr/local/lib64/libpangoft2-1.0.so.0: undefined reference to `hb_ft_face_create'
/usr/local/lib64/gcc/x86_64-cros-linux-gnu/8.3.0/../../../../x86_64-cros-linux-gnu/bin/ld: /usr/local/lib64/libpangoft2-1.0.so.0: undefined reference to `hb_glib_script_from_script'
/usr/local/lib64/gcc/x86_64-cros-linux-gnu/8.3.0/../../../../x86_64-cros-linux-gnu/bin/ld: /usr/local/lib64/gcc/x86_64-cros-linux-gnu/8.3.0/../../../../lib64/libgtk-3.so: undefined reference to `hb_ft_font_create'
/usr/local/lib64/gcc/x86_64-cros-linux-gnu/8.3.0/../../../../x86_64-cros-linux-gnu/bin/ld: /usr/local/lib64/gcc/x86_64-cros-linux-gnu/8.3.0/../../../../lib64/libpango-1.0.so: undefined reference to `hb_glib_script_to_script'
/usr/local/lib64/gcc/x86_64-cros-linux-gnu/8.3.0/../../../../x86_64-cros-linux-gnu/bin/ld: /usr/local/lib64/gcc/x86_64-cros-linux-gnu/8.3.0/../../../../lib64/libgtk-3.so: undefined reference to `hb_ft_font_set_funcs'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:208: qemu-system-aarch64] Error 1
make[1]: Leaving directory '/usr/local/tmp/crew/qemu-4.2.0.tar.xz.dir/qemu-4.2.0/aarch64-softmmu'
make: *** [Makefile:491: aarch64-softmmu/all] Error 2
qemu failed to install: `make V=0 -j4` exited with 2

block/ssh.c: In function ‘check_host_key_knownhosts’:
block/ssh.c:301:48: error: ‘SSH_PUBLICKEY_HASH_SHA256’ undeclared (first use in this function); did you mean ‘SSH_PUBLICKEY_HASH_SHA1’?
             r = ssh_get_publickey_hash(pubkey, SSH_PUBLICKEY_HASH_SHA256,
                                                ^~~~~~~~~~~~~~~~~~~~~~~~~
                                                SSH_PUBLICKEY_HASH_SHA1
block/ssh.c:301:48: note: each undeclared identifier is reported only once for each function it appears in
block/ssh.c:307:27: warning: implicit declaration of function ‘ssh_get_fingerprint_hash’; did you mean ‘ssh_get_pubkey_hash’? [-Wimplicit-function-declaration]
             fingerprint = ssh_get_fingerprint_hash(SSH_PUBLICKEY_HASH_SHA256,
                           ^~~~~~~~~~~~~~~~~~~~~~~~
                           ssh_get_pubkey_hash
block/ssh.c:307:27: warning: nested extern declaration of ‘ssh_get_fingerprint_hash’ [-Wnested-externs]
make: *** [/usr/local/tmp/crew/qemu-4.2.0.tar.xz.dir/qemu-4.2.0/rules.mak:69: block/ssh.o] Error 1
make: *** Waiting for unfinished jobs....
qemu failed to install: `make V=0 -j4` exited with 2

--- a/packages/libssh.rb
+++ b/packages/libssh.rb
@@ -3,34 +3,26 @@ require 'package'
 class Libssh < Package
   description 'libssh is a multiplatform C library implementing the SSHv2 and SSHv1 protocol on client and server side.'
   homepage 'https://www.libssh.org/'
-  version '0.8.1'
+  version '0.9.5'
   compatibility 'all'
-  source_url 'https://www.libssh.org/files/0.8/libssh-0.8.1.tar.xz'
-  source_sha256 'd17f1267b4a5e46c0fbe66d39a3e702b8cefe788928f2eb6e339a18bb00b1924'
-
-  binary_url ({
-    aarch64: 'https://dl.bintray.com/chromebrew/chromebrew/libssh-0.8.1-chromeos-armv7l.tar.xz',
-     armv7l: 'https://dl.bintray.com/chromebrew/chromebrew/libssh-0.8.1-chromeos-armv7l.tar.xz',
-       i686: 'https://dl.bintray.com/chromebrew/chromebrew/libssh-0.8.1-chromeos-i686.tar.xz',
-     x86_64: 'https://dl.bintray.com/chromebrew/chromebrew/libssh-0.8.1-chromeos-x86_64.tar.xz',
-  })
-  binary_sha256 ({
-    aarch64: '0928c31849b027b3188b765c4ee7434cad046523cf4a54355e125421baaf64c5',
-     armv7l: '0928c31849b027b3188b765c4ee7434cad046523cf4a54355e125421baaf64c5',
-       i686: 'ee40bfa70d6b0fd644ab58f081dd3c82f29ca0f28586698eed55c79d79656c47',
-     x86_64: '8cca9b0403091a838e65b2306cb5bb3b8ed549502d7e54c442af0fac1a3fc671',
-  })
+  source_url 'https://www.libssh.org/files/0.9/libssh-0.9.5.tar.xz'
+  source_sha256 'acffef2da98e761fc1fd9c4fddde0f3af60ab44c4f5af05cd1b2d60a3fa08718'
 
   depends_on 'libgcrypt'
-
+    
+  def self.patch
+      `echo "$(sed '/\# global needed variable/a set\(CMAKE_C_FLAGS \"-fstack-protector-strong\"\)~\
+add_compile_options\(-fstack-protector-strong\)~\
+link_libraries\(-fstack-protector-strong\)' CMakeLists.txt)" > ./CMakeLists.txt`
+      `echo "$(tr '~' '\n' < CMakeLists.txt)" > CMakeLists.txt`
+  end
   def self.build
-    FileUtils.mkdir 'build'
+    Dir.mkdir 'build'
     Dir.chdir 'build' do
       system "cmake \
+             #{CREW_CMAKE_OPTIONS} \
              -DWITH_GCRYPT=ON \
-             -DCMAKE_INSTALL_PREFIX=#{CREW_PREFIX} \
-             -DLIB_INSTALL_DIR=#{CREW_LIB_PREFIX} \
-             -DCMAKE_BUILD_TYPE=Release .."
+             .."
       system 'make'
     end
   end

--- a/packages/libssh.rb
+++ b/packages/libssh.rb
@@ -10,17 +10,12 @@ class Libssh < Package
 
   depends_on 'libgcrypt'
     
-  def self.patch
-      `echo "$(sed '/\# global needed variable/a set\(CMAKE_C_FLAGS \"-fstack-protector-strong\"\)~\
-add_compile_options\(-fstack-protector-strong\)~\
-link_libraries\(-fstack-protector-strong\)' CMakeLists.txt)" > ./CMakeLists.txt`
-      `echo "$(tr '~' '\n' < CMakeLists.txt)" > CMakeLists.txt`
-  end
   def self.build
     Dir.mkdir 'build'
     Dir.chdir 'build' do
       system "cmake \
              #{CREW_CMAKE_OPTIONS} \
+             -DCMAKE_C_FLAGS='-fstack-protector-strong' \
              -DWITH_GCRYPT=ON \
              .."
       system 'make'

